pilot/pkg/networking/core/v1alpha3/route/route_cache.go

```diff
@@ -96,7 +97,12 @@ func (r *Cache) DependentConfigs() []model.ConfigHash {
 		}.HashCode())
 	}
 	for _, vs := range r.VirtualServices {
-		configs = append(configs, model.ConfigKey{Kind: kind.VirtualService, Name: vs.Name, Namespace: vs.Namespace}.HashCode())
+		if model.UseGatewaySemantics(vs) {
```

Comment:
Do we miss this for Ingress?

Reply:
AFAIK rds cache only applies to sidecars, so we don't need to care about ingress
But for gateway-api we are supporting **mesh** routes applying to sidecars